cfme/configure/access_control.py

```diff
@@ -168,149 +238,298 @@ def description(self):
+                   'username': self.ldap_credentials.principal,
+                   'password': self.ldap_credentials.secret})
+        view.retrieve_button.click()
+        return view
 
     def _retrieve_ext_auth_user_groups(self):
-        navigate_to(self, 'Add')
-        fill(self.group_form, {'lookup_ldap_groups_chk': True,
-                               'user_to_look_up': self.user_to_lookup,
-                               },)
-        sel.wait_for_element(form_buttons.retrieve)
-        sel.click(form_buttons.retrieve)
+        view = navigate_to(self, 'Add')
+        view.fill({'lookup_ldap_groups_chk': True,
+                   'user_to_look_up': self.user_to_lookup})
+        view.retrieve_button.click()
+        return view
+
+    def _fill_ldap_group_lookup(self, view):
+        view.fill({'ldap_groups_for_user': self.description,
+                   'description_txt': self.description,
+                   'role_select': self.role,
+                   'group_tenant': self.tenant})
+        view.add_button.click()
+        view = self.create_view(AllGroupView)
+        view.flash.assert_success_message('Group "{}" was saved'.format(self.description))
+        assert view.is_displayed
 
     def add_group_from_ldap_lookup(self):
-        self._retrieve_ldap_user_groups()
-        fill(self.group_form, {'ldap_groups_for_user': self.description,
-                               'description_txt': self.description,
-                               'role_select': self.role,
-                               'group_tenant': self.tenant,
-                               },
-             action=form_buttons.add)
-        flash.assert_success_message('Group "{}" was saved'.format(self.description))
+        view = self._retrieve_ldap_user_groups()
+        self._fill_ldap_group_lookup(view)
 
     def add_group_from_ext_auth_lookup(self):
-        self._retrieve_ext_auth_user_groups()
-        fill(self.group_form, {'ldap_groups_for_user': self.description,
-                               'description_txt': self.description,
-                               'role_select': self.role,
-                               'group_tenant': self.tenant,
-                               },
-             action=form_buttons.add)
-        flash.assert_success_message('Group "{}" was saved'.format(self.description))
+        view = self._retrieve_ext_auth_user_groups()
+        self._fill_ldap_group_lookup(view)
 
     def update(self, updates):
-        navigate_to(self, 'Edit')
-        fill(self.group_form, {'description_txt': updates.get('description'),
-                               'role_select': updates.get('role'),
-                               'group_tenant': updates.get('tenant')},
-             action=form_buttons.save)
-        flash.assert_success_message(
-            'Group "{}" was saved'.format(updates.get('description', self.description)))
+        view = navigate_to(self, 'Edit')
+        changed = view.fill({
```

Comment:
As per above, there is no way to update the tags from the update method and typically we like create/update to perform the same sets of operations.